envconfig.go

```diff
@@ -517,6 +517,15 @@ func lookup(key string, opts *options, l Lookuper) (string, bool, bool, error) {
 			// Expand the default value. This allows for a default value that maps to
 			// a different variable.
 			val = os.Expand(opts.Default, func(i string) string {
+				// special case for the name of the env var itself, including
+				// its prefix
+				if i == "@" {
```

Comment:
Thanks for thinking of edge cases!

I think that having an env var like `@_FOO` is a bit esoteric and probably not something often encountered in the wild.  As currently implemented, `$@_FOO` would expand just the `$@` part (I believe that this is the same behavior as POSIX shell, or at the very least Bash and ZSH).  If using `@_FOO` as the env var name were indeed intended, then it could be made unambiguous by writing `${@_FOO}` instead.

Perhaps a more unambiguous-by-default, single-letter variable like underscore (`_`) would be a better choice.  The [implementation does not treat `_` as a special, single-character shell variable][1].  The env var name used for expansion will be the longest alphanumeric match (e.g., `$_FOO BAR` will use `_FOO`).  Moreover, it might be more familiar for Bash and ZSH users, because `$_` is expanded as something akin to "the last thing" (not entirely a correct statement 😇), and they also perform longest alphanumeric matching when underscore follows the dollar sign.  So, underscore might be more intuitive in that respect.

Here are some example expansions off the top of my head:

- `$_FOO` expands to the value of the env var `_FOO`
- `${_}FOO` expands to the actual env var name followed by `FOO`
- `$_` just expands to the actual env var name

When the `_` is found, it _could_ first perform an actual lookup for the env var `_` (or its prefixed version) and, iff that is not defined, expand to the actual env var name.  This functionality might be getting into the territory of being potentially too much magic; but, I think it makes sense.

Some questions are:
- What is the likelihood of having an env var precisely named `_`?
- What is the likelihood of having env vars prefixed with `@`?

For the sake of discussion, I will modify this pull request to replace `@` with `_`, and use the potentially-too-much-magic functionality which looks up `_` before defaulting to the actual env var name, as mentioned above.

[1]: https://github.com/golang/go/blob/8c92897e15d15fbc664cd5a05132ce800cf4017f/src/os/env.go#L54-L62